install/kubernetes/cilium/charts/operator/templates/deployment.yaml

```diff
@@ -33,7 +33,13 @@ spec:
         - --config-dir=/tmp/cilium/config-map
         - --debug=$(CILIUM_DEBUG)
         command:
-        - cilium-operator
+{{- if .Values.global.eni }}
```

Comment:
The image should pick the correct binary as default command, but if we prefer this to be explicit I won't oppose :)

Reply:
If neither of the provider flags is set, the default else condition will use cilium-operator-generic as the command